db/db_impl/db_impl_open.cc

```diff
@@ -1697,6 +1755,9 @@ Status DBImpl::Open(const DBOptions& db_options, const std::string& dbname,
         }
       }
     }
+    if (s.ok()) {
+      s = impl->LogAndApply(&recovery_version_edits);
```

Comment:
iiuc, this should happen after the subsequent WAL sync? Is this the issue you mentioned before?

Reply:
Yes, It seems like `LogAndApply` needs to be called before `InstallSuperVersionAndScheduleWork` otherwise it fails to find the keys. It didn't get the chance to look into `InstallSuperVersionAndScheduleWork` more but the failing was because of the order of API calls.

Reply:
Can you try switching the order of writing and syncing WAL with installing superversions as https://github.com/riversand963/rocksdb/blob/tmp-9634/db/db_impl/db_impl_open.cc#L1798:L1810?
Then you can do
- Write and sync WAL
- LogAndApply
- InstallSuperVersions

Reply:
Sure. Let me try this one.